opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -626,147 +649,152 @@ static int mca_btl_base_am_rdma_respond(mca_btl_base_module_t *btl,
-                                   struct mca_btl_base_registration_handle_t *local_handle,
-                                   void *context, void *cbdata, int status)
+am_rdma_rdma_complete(mca_btl_base_module_t *btl,
+                      struct mca_btl_base_endpoint_t *endpoint, void *local_address,
+                      struct mca_btl_base_registration_handle_t *local_handle,
+                      void *context, void *cbdata, int status)
 {
-    mca_btl_base_rdma_operation_t *operation = (mca_btl_base_rdma_operation_t *) context;
+    am_rdma_operation_t *operation = (am_rdma_operation_t *) context;
 
     BTL_VERBOSE(("BTL RDMA operation complete. status=%d", status));
 
     assert(OPAL_SUCCESS == status);
 
     operation->is_completed = true;
-    int ret = mca_btl_base_am_rdma_respond(operation->btl, operation->endpoint,
-                                           &operation->descriptor, NULL, &operation->hdr);
+    int ret = am_rdma_respond(operation->btl, operation->endpoint,
+                              &operation->descriptor, NULL, &operation->hdr);
     if (OPAL_UNLIKELY(OPAL_SUCCESS != ret)) {
         BTL_VERBOSE(
             ("could not send a response. queueing the response for later. endpoint=%p, ret=%d",
              (void*) endpoint, ret));
-        mca_btl_base_rdma_queue_operation(btl, NULL, NULL, 0, NULL, operation);
+        am_rdma_queue_operation(btl, NULL, NULL, 0, NULL, operation);
     }
 
     OBJ_RELEASE(operation);
 }
 
-static int mca_btl_base_am_rdma_target_get(mca_btl_base_module_t *btl,
-                                           struct mca_btl_base_endpoint_t *endpoint,
-                                           mca_btl_base_descriptor_t **descriptor,
-                                           void *target_address, const mca_btl_base_rdma_hdr_t *hdr,
-                                           mca_btl_base_rdma_operation_t **operation)
+
+static int am_rdma_target_put(mca_btl_base_module_t *btl,
+                              struct mca_btl_base_endpoint_t *endpoint,
+                              mca_btl_base_descriptor_t **descriptor,
+                              const mca_btl_base_segment_t *segments,
+                              size_t segment_count, void *target_address,
+                              const am_rdma_hdr_t *hdr,
+                              am_rdma_operation_t **operation)
 {
     if (hdr->data.rdma.use_rdma) {
         if (NULL == *operation) {
-            *operation = mca_btl_base_rdma_alloc_operation(btl, endpoint, *descriptor, hdr);
+            *operation = am_rdma_alloc_operation(btl, endpoint, *descriptor, hdr);
             if (NULL == *operation) {
                 return OPAL_ERR_OUT_OF_RESOURCE;
             }
         }
 
         /* btl supports put but not get. emulating get with put */
         OBJ_RETAIN(*operation);
-        int ret = btl->btl_put(
+        int ret = btl->btl_get(
```

Comment:
using `btl_get` here inside of `am_rdma_target_put` is contrary to the comment on line 695.

Reply:
That looks like Nathan cut-n-pasted the comment from the target_get() code (which is probably what he implemented first).  I fixed the comment (the code was correct).